catalyst/loggers/comet.py

```diff
@@ -0,0 +1,254 @@
+            # ...
+
+        runner = CustomRunner().run()
+    """
+
+    def __init__(
+        self,
+        workspace: str = None,
+        project_name: str = None,
+        experiment_id: str = None,
+        comet_mode: str = "online",
+        tags: List = None,
+        logging_frequency: int = 10,
+        experiment_kwargs: dict = {},
+    ) -> None:
+        self.workspace = workspace
+        self.project_name = project_name
+        self.experiment_id = experiment_id
+        self.experiment_kwargs = experiment_kwargs
+        self.comet_mode = comet_mode
+        self.logging_frequency = logging_frequency
+
+        self.experiment = self._get_experiment(self.comet_mode, self.experiment_id)
+        self.experiment.log_other("Created from", "Catalyst")
+        if tags is not None:
+            self.experiment.add_tags(tags)
+    
+    def _get_experiment(self, mode, experiment_id=None):
+        if mode == "offline":
+            if experiment_id is not None:
+                return comet_ml.ExistingOfflineExperiment(
+                    previous_experiment=experiment_id,
+                    workspace=self.workspace,
+                    project_name=self.project_name,
+                    **self.experiment_kwargs,
+                )
+
+            return comet_ml.OfflineExperiment(
+                workspace=self.workspace,
+                project_name=self.project_name,
+                **self.experiment_kwargs,
+            )
+
+        else:
+            if experiment_id is not None:
+                return comet_ml.ExistingExperiment(
+                    previous_experiment=experiment_id,
+                    workspace=self.workspace,
+                    project_name=self.project_name,
+                    **self.experiment_kwargs,
+                )
+
+            return comet_ml.Experiment(
+                workspace=self.workspace,
+                project_name=self.project_name,
+                **self.experiment_kwargs,
+            )
+
+    def log_metrics(
+        self,
+        metrics: Dict[str, float],
+        scope: str = None,
+        # experiment info
+        run_key: str = None,
+        global_epoch_step: int = 0,
+        global_batch_step: int = 0,
+        global_sample_step: int = 0,
+        # stage info
+        stage_key: str = None,
+        stage_epoch_len: int = 0,
+        stage_epoch_step: int = 0,
+        stage_batch_step: int = 0,
+        stage_sample_step: int = 0,
+        # loader info
+        loader_key: str = None,
+        loader_batch_len: int = 0,
+        loader_sample_len: int = 0,
+        loader_batch_step: int = 0,
+        loader_sample_step: int = 0,
+    ) -> None:
+        """Logs the metrics to the logger."""
+
+        if global_batch_step % self.logging_frequency == 0:
+            self.experiment.log_metrics(
+                metrics,
+                step=global_batch_step,
+                epoch=global_epoch_step,
+                prefix=f"{stage_key}/{scope}",
```

Comment:
I would make this `f"{stage_key}/{loader_key}_{scope}"` since the `loader_key` tells you which dataset (train, valid or test) the metric was computed on. You can have cases where the `stage_key` is train but the ` loader_key` is `valid`. With the current implementation, these metrics will get overwritten.

Reply:
In all my tests it seemed like the stage_key and the loader_key were all set on "train" so it gave the metric a weird name. You can see an example in this ticket: https://comet-ml.atlassian.net/browse/CM-781

Reply:
So I think it's okay if the name looks weird like `train/train_batch_precision_09` It would be up to the user to change the stage_key to something that is more relevant.